sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -1742,6 +1742,51 @@ class MapboxMap :
     }
   }
 
+  /**
+   * Adds a model to be used in the style. This API can also be used for updating
+   * a model. If the model for a given `modelId` was already added, it gets replaced by the new model.
+   *
+   * The model can be used in `model-id` property in model layer.
+   *
+   * @param modelId An identifier of the model.
+   * @param modelUri A URI for the model.
+   */
+  @MapboxExperimental
+  fun addStyleModel(modelId: String, modelUri: String) {
+    checkNativeMap("addStyleModel")
+    val expected = nativeMap.addStyleModel(modelId, modelUri)
+    if (expected.isError) {
+      logE(TAG, expected.error ?: "Error occurred in MapboxMap.addStyleModel!")
+    }
+  }
+
+  /**
+   * Removes a model from the style.
+   *
+   * @param modelId The identifier of the model to remove.
+   */
+  @MapboxExperimental
+  fun removeStyleModel(modelId: String) {
+    checkNativeMap("removeStyleModel")
+    val expected = nativeMap.removeStyleModel(modelId)
+    if (expected.isError) {
+      logE(TAG, expected.error ?: "Error occurred in MapboxMap.removeStyleModel!")
+    }
+  }
+
+  /**
+   * Checks whether a model exists.
+   *
+   * @param modelId The identifier of the model.
+   *
+   * @return True if model exists, false otherwise.
+   */
+  @MapboxExperimental
+  fun hasStyleModel(modelId: String): Boolean {
+    checkNativeMap("hasStyleModel")
+    return nativeMap.hasStyleModel(modelId)
+  }
```

Comment:
aren't these supposed to be style APIs? why do we add them to the MapboxMap?

Reply:
I decided based on our experience with map projection that better expose it on map as well allowing to actually add 3D models before style is loaded then leaving the only opportunity to use runtime styling and introduce visual delay. As it's experimental - we may drop this API if needed later.

Reply:
I'm not in favour in introducing duplicated APIs, unless we find a solid issue or use case that couldn't be worked around, otherwise we are just keep making our API surface more and more messy.

Reply:
discussed offline, leaving as is for now